test/core.jl

```diff
@@ -123,18 +123,18 @@ end
     @test convert(ImageMeta{Gray}, A) == A
 end
 
-@testset "reinterpret" begin
-    # It's possible that reinterpret shouldn't be defined for ImageMeta, but...
-    A = rand(Float32, 4, 5)
-    M = ImageMeta(A, meta=true)
-    Mr = reinterpret(Gray, M)
-    @test eltype(Mr) == Gray{Float32}
-    @test Mr["meta"] = true
-    # Ensure that it never gets defined for the un-reinterpretable
-    M = ImageMeta(view(A, 1:2:3, 1:4), meta=true)
-    @test_throws ErrorException reinterpret(Gray, M)
-    @test_throws ErrorException reinterpret(Gray{Float32}, M)
-end
+# @testset "reinterpret" begin
```

Comment:
These would all become `reinterpretc` methods if we want to keep them around.

The reason to consider not having them at all is that ImageMeta includes the Dict, and so `reinterpret` applies to only part of the overall object. On the flip side, if we support `channelview` and similar then it's basically `reinterpretc` anyway.